app/client/utils.js

```diff
@@ -306,6 +319,158 @@ export const processFiltersForQuery = filters => {
   return encoded_filters;
 };
 
+export const parseEncodedParams = (allowedFilters, params) => {
+  let metric_types = ["credit_transfer", "user"];
+  let filters = [];
+  let param_array;
+  let allowedValues;
+  let gt;
+  let lt;
+  let eq;
+  try {
+    if (allowedFilters && params) {
+      param_array = params.split(":");
+      param_array.map((param, i) => {
+        let filter = {};
+        let filterAttribute = param.split("(")[0];
+        filter["type"] = allowedFilters[filterAttribute].type;
+        if (
+          USER_FILTER_TYPE.DISCRETE === filter.type ||
+          USER_FILTER_TYPE.BOOLEAN_MAPPING === filter.type
+        ) {
+          allowedValues = param.split("(IN)")[1];
+          filter["allowedValues"] = allowedValues
+            .substr(1, allowedValues.length - 2)
+            .split(",");
+          filter["attribute"] = filterAttribute;
+          filter["id"] = i + 1;
+        } else {
+          gt = param.split("(GT)");
+          lt = param.split("(LT)");
+          eq = param.split("(EQ)");
+          if (gt.length > 1) {
+            allowedValues = gt[1];
+            filter["type"] = ">";
+            filter["threshold"] = allowedValues.substr(
+              1,
+              allowedValues.length - 2
+            );
+          } else if (lt.length > 1) {
+            allowedValues = lt[1];
+            filter["type"] = "<";
+            filter["threshold"] = allowedValues.substr(
+              1,
+              allowedValues.length - 2
+            );
+          } else {
+            allowedValues = eq[1];
+            filter["type"] = "=";
+            filter["threshold"] = allowedValues.substr(
+              1,
+              allowedValues.length - 2
+            );
+          }
+          filter["attribute"] = filterAttribute;
+          filter["id"] = i + 1;
+        }
+
+        filters.push(filter);
+      });
+    }
+  } catch (e) {
+    console.log("Something went wrong", e);
+    allowedFilters = null;
+    metric_types.map(mt => {
+      params = {
+        disable_cache: true,
+        group_by: "ungrouped",
+        metric_type: mt
+      };
+      store.dispatch(LoadMetricAction.loadMetricRequest({ query: params }));
```

Comment:
feels like a kinda strange place to have side effects to me. Is it unweildly to have the catch somewhere else to keep this function pure?